addons/l10n_fr_certification/models/account_closure.py

```diff
@@ -0,0 +1,153 @@
+from datetime import datetime, timedelta
+
+from openerp import models, api, fields
+from openerp.fields import Datetime as FieldDateTime
+from openerp.tools.translate import _
+from openerp.exceptions import UserError
+
+WRITE_MSG = _('Sale Closings are not meant to be written or deleted under any circumstances.')
+
+
+class AccountClosure(models.Model):
+    _name = 'account.sale.closure'
+
+    name = fields.Char(help="Frequency and unique sequence number")
+    company_id = fields.Many2one('res.company', string='Company', readonly=True)
+    date_closure_stop = fields.Datetime(string="Closing Date", help='Date to which the values are computed', readonly=True)
+    date_closure_start = fields.Datetime(string="Starting Date", help='Date from which the total interval is computed', readonly=True)
+    frequency = fields.Selection(string='Closing Type', selection=[('daily', 'Daily'), ('monthly', 'Monthly'), ('annually', 'Annual')], readonly=True)
+    total_interval = fields.Monetary(string="Period Total", help='Total in receivable accounts during the interval, excluding overlapping periods', readonly=True)
+    total_beginning = fields.Monetary(string="Cumulative Grand Total", help='Total in receivable accounts since the beginning of times', readonly=True)
+    sequence_number = fields.Integer('Sequence #', readonly=True)
+    move_ids = fields.Many2many('account.move', string='Journal entries that are included in the computation', readonly=True)
+    currency_id = fields.Many2one('res.currency', string='Currency', help="The company's currency", readonly=True)
+
+    def _query_closures_for_move_ids(self, frequency, company):
+        query = '''SELECT rel.account_move_id as move_id
+                    FROM account_move_account_sale_closure_rel rel
+                    JOIN account_sale_closure clo on rel.account_sale_closure_id = clo.id
+                    JOIN account_move m on rel.account_move_id = m.id
+                    WHERE clo.frequency = %(frequency)s
+                        AND clo.company_id = %(company_id)s
+                        AND m.company_id = %(company_id)s
+                        AND m.state = 'posted' '''
+
+        self.env.cr.execute(query, {'frequency': frequency, 'company_id': company.id})
+        return [res['move_id'] for res in self.env.cr.dictfetchall()]
+
+    def _build_query_for_aml(self, company, date_start='', date_stop='', avoid_move_ids=[]):
+        params = {'company_id': company.id}
+        query = '''SELECT m.write_date AS move_write_date,
+                          m.id AS move_id,
+                          debit-credit AS balance
```

Comment:
there is a balance field, already stored on the account_move_line object. Save you some computation time